python/src/iceberg/expression/literals.py

```diff
@@ -0,0 +1,741 @@
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import datetime
+import uuid
+from decimal import ROUND_HALF_UP, Decimal
+from functools import singledispatch
+
+try:
+    from functools import singledispatchmethod
+except ImportError:
+    from singledispatch import singledispatchmethod  # type: ignore
```

Comment:
I think it'd be helpful if we're more deliberate here that we're falling back to the backport for versions below 3.8, that way in the future when we drop python 3.7 support it will be obvious that this line can be removed. How about:
```py
import sys

if sys.version_info >= (3, 8):
    from functools import singledispatchmethod
else:
    from singledispatch import singledispatchmethod
```